qemu/tests/timerdevice_boot.py

```diff
@@ -158,23 +185,42 @@ def verify_guest_clock_source(session, expected):
             time.sleep(sleep_time)
 
         session = vm.reboot()
+
         error.context("Check the system time on guest and host", logging.info)
         (host_time, guest_time) = utils_test.get_time(session, time_command,
                                                       time_filter_re, time_format)
-        drift = abs(float(host_time) - float(guest_time))
-        if drift > timerdevice_drift_threshold:
-            raise error.TestFail("The guest's system time is different with"
-                                 " host's. Host time: '%s', guest time:"
-                                 " '%s'" % (host_time, guest_time))
-
-        get_hw_time_cmd = params.get("get_hw_time_cmd")
-        if get_hw_time_cmd:
-            error.context(
-                "Check the hardware time on guest and host", logging.info)
-            host_time = utils.system_output(get_hw_time_cmd)
-            guest_time = session.cmd(get_hw_time_cmd)
+        if params.get("rtc_base") == "2006-6-17":
+            if "rhel7" in guest_name:
+                error.context("Unmask chronyd service")
+                cmd = "systemctl unmask chronyd"
+                session.cmd_status_output(cmd)
+            guest_time = time.strftime('%Y-%m-%d %H', time.localtime(guest_time))
+            os_type = params["os_type"]
+            if os_type == "linux":
```

Comment:
Thanks @CongLi 
Merge the duplicated code to a function "verify_base_date_system_time"